### PR TITLE
Added dragMouse function

### LINKED ISSUE #127: selecting text with mouse commands
I have some code like this:

```js
    robot.mouseToggle("down")
    robot.moveMouse(data.x, data.y)
    robot.mouseToggle("up")
```

I was hoping this would have the effect of selecting text, but I can't get the text to get highlighted Any ideas?

### PR DESCRIPTION
**Pull request by noonat**
This adds a new dragMouse function that allows you to do things like:

```
robot.moveMouse(300, 300);
robot.mouseToggle('down');
robot.dragMouse(400, 300);
robot.mouseToggle('up');
```

It just calls moveMouse for platforms other than OS X.

Resolves #127.

### CODE REVIEW

File: src/mouse.c [hunk 1]
```diff
@@ -18,6 +18,44 @@
...
+	((button) == (LEFT_BUTTON) ? kCGEventLeftMouseDown \
+	                       : ((button) == RIGHT_BUTTON ? kCGEventRightMouseDown \
+	                                                   : kCGEventOtherMouseDown))
+
+#define MMMouseUpToCGEventType(button) \
+	((button) == LEFT_BUTTON ? kCGEventLeftMouseUp \
+	                         : ((button) == RIGHT_BUTTON ? kCGEventRightMouseUp \
+	                                                     : kCGEventOtherMouseUp))
+
+#define MMMouseDragToCGEventType(button) \
```
Comment (noonat) [inline src/mouse.c]: Moved these up here so I can make use of this new macro.

File: src/robotjs.cc [hunk 2]
```diff
@@ -26,7 +26,66 @@ int keyboardDelay = 10;
 
 */
 
-NAN_METHOD(moveMouse) 
+int CheckMouseButton(const char * const b, MMMouseButton * const button)
```
Comment (noonat) [inline src/robotjs.cc]: Added this new function, following the style of your CheckKeyCodes, and changed the other mouse functions to also use it.

**Comment by octalmage** [comment]
This is so awesome, thanks! 



I'll get this into the next release.

File: src/robotjs.cc (hunk 2 above)
Comment (octalmage) [inline src/robotjs.cc]: Good call!

**Comment by maxogden** [comment]
pretty excited for this :)

**Comment by octalmage** [comment]
Looks good! I'll get this out to npm shortly.

**Comment by octalmage** [comment]
RobotJS v0.3.5 has been published to npm and it includes this pull request! Thanks again @noonat.

**Pull request merged**